components/script/dom/document.rs

```diff
@@ -5051,6 +5051,29 @@ impl Document {
         self.image_animation_manager.borrow_mut()
     }
 
+    pub(crate) fn update_image_animations(&self) {
+        let mut image_animation_manager = self.image_animation_manager.borrow_mut();
+        image_animation_manager
+            .update_active_frames(&self.window, self.current_animation_timeline_value());
+        if !self.animations().animations_present() {
+            let next_scheduled_time =
+                image_animation_manager.next_schedule_time(self.current_animation_timeline_value());
```

Comment:
I think I would have `next_schedule_time` return an `Option` and only schedule a timer if it is not `None`. This will allow you to gracefully remove the `unwrap()` (which are a bit scary) in the method.

```suggestion
            let next_scheduled_time =
                image_animation_manager.next_schedule_time(self.current_animation_timeline_value());
```

Reply:
Make sense!